### PR TITLE
Reuse templatetools.JSONObject during the download process

### PR DESCRIPTION
**Pull request by jskelin**
…for the automation

#### What this PR does / Why we need it:
Harmonize code for bucket and automation download. Because the functionality is similar, it is easier to have code also that looks similar.

### CODE REVIEW

**Comment by Laubi** [comment]
This might be controversial, however, I'd rather not introduce the `templatetools.NewJSONObject(o)` to handle this case. 
In my personal opinion, I think it adds unnecessary overhead to the simple handling of the `map[k]v` structure. I personally have a harder time to read the object-oriented code than the existing one

File: pkg/download/bucket/bucket.go [hunk 1]
```diff
@@ -77,14 +76,15 @@ func (d *Downloader) convertAllObjects(projectName string, objects [][]byte) []c
 
 		// exclude builtin bucket names
 		if tools.IsDefault(c.OriginObjectId) {
-			log.Debug("Skipping download of immutable default Bucket %s", c.OriginObjectId)
+			log.Debug("Skipping download of immutable default Bucket %s", c.OriginObjectId) //TODO: WithFields??
```
Comment (UnseenWizzard) [inline pkg/download/bucket/bucket.go:79]: IMO wouldn't hurt to include the ID as a field.

**Comment by warber** [comment]
@jskelin Can we close this?

**Comment by jskelin** [comment]
abandoned

**Pull request closed without merging**